src/pages/ReportDetailsPage.js

```diff
@@ -76,6 +77,18 @@ function ReportDetailsPage(props) {
 
     const isGroupDMChat = useMemo(() => ReportUtils.isDM(props.report) && participants.length > 1, [props.report, participants.length]);
 
+    const isPrivateNotesFetchTriggered = !_.isUndefined(props.report.isLoadingPrivateNotes);
+
+    useEffect(() => {
+        // Do not fetch private notes if isLoadingPrivateNotes is already defined, or if network is offline.
+        if (isPrivateNotesFetchTriggered || props.network.isOffline) {
+            return;
+        }
+
+        Report.getReportPrivateNote(props.report.reportID);
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- do not add report.isLoadingPrivateNotes to dependencies
+    }, [props.report.reportID, props.network.isOffline, isPrivateNotesFetchTriggered]);
+
```

Comment:
This change based on the existed logic:
https://github.com/Expensify/App/blob/63014263a542633621da9efd7617b5f9d0879fcc/src/pages/home/report/withReportAndPrivateNotesOrNotFound.js#L60-L68